### ISSUE TITLE
no such file or directory: libzf.so

### ISSUE THREAD

**Issue by dchinmay2** [natecraddock/telescope-zf-native.nvim#1]
Love your algorithm, sometimes fzf gets very annoying. I'm already getting better matches in the terminal!
The zf repo generates an executable, how do I build the library?

**Comment by natecraddock** [comment]
Hi! Thanks for the interest, and glad to hear zf is working well for others!

So I threw this together on a night about a week ago and haven't touched it much since then because I was finishing up my sessions and workspaces plugins. That means I haven't gotten around to making it easy to install, sorry about that! I made plans today to tidy up this repo this week (and fix a few remaining problems) so hopefully later this week it should be more ready to go.

Until then, this is the command I used to build. Run from the root of the repo after pulling in the submodule.

```
zig build-lib zf/src/filter.zig -O ReleaseFast -dynamic -femit-bin=lib/libzf.so
```

This just builds the filter algorithm on its own. The one big problem at the moment is that filenames aren't actually prioritized in the connection from telescope to zf. I'll address that later this week too.

Also, if you have any feedback on zf (cases you wish matched better, bugs, other ideas for scoring) I would be happy to hear it!
